azurecompute/src/test/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapterLiveTest.java

```diff
@@ -0,0 +1,139 @@
+      adapter = injector.getInstance(AzureComputeServiceAdapter.class);
+      templateBuilder = injector.getInstance(TemplateBuilder.class);
+      sshFactory = injector.getInstance(SshClient.Factory.class);
+      return injector.getInstance(AzureComputeApi.class);
+   }
+
+   @Test
+   public void testListLocations() {
+      Iterable<Location> locations = adapter.listLocations();
+      assertFalse(Iterables.isEmpty(locations), "locations must not be empty");
+   }
+
+   @Test
+   public void testListImages() {
+      assertFalse(Iterables.isEmpty(adapter.listImages()), "images must not be empty");
+   }
+
+   private static final PrioritizeCredentialsFromTemplate prioritizeCredentialsFromTemplate = new PrioritizeCredentialsFromTemplate(
+         new DefaultCredentialsFromImageOrOverridingCredentials());
+
+   @Test
+   public void testCreateNodeWithGroupEncodedIntoNameThenStoreCredentials() {
+      String group = "foo";
+      String name = "node" + new Random().nextInt();
+
+      Template template = templateBuilder.build();
+
+      deployment = adapter.createNodeWithGroupEncodedIntoName(group, name, template);
+      assertEquals(deployment.getNode().name(), name);
+      assertEquals(deployment.getNodeId(), deployment.getNode().name());
+      assert InetAddresses.isInetAddress(deployment.getNode().virtualIPs().get(0).address()) : deployment;
+      doConnectViaSsh(deployment.getNode(), prioritizeCredentialsFromTemplate.apply(template, deployment.getCredentials()));
+   }
+
+   protected void doConnectViaSsh(Deployment deployment, LoginCredentials creds) {
+      SshClient ssh = sshFactory.create(HostAndPort.fromParts(deployment.virtualIPs().get(0).address(), 22), creds);
+      try {
+         ssh.connect();
+         ExecResponse hello = ssh.exec("echo hello");
+         assertEquals(hello.getOutput().trim(), "hello");
+         System.err.println(ssh.exec("df -k").getOutput());
+         System.err.println(ssh.exec("mount").getOutput());
+         System.err.println(ssh.exec("uname -a").getOutput());
+      } finally {
+         if (ssh != null)
+            ssh.disconnect();
+      }
+   }
+
+   @Test
+   public void testListHardwareProfiles() {
+      Iterable<RoleSize> roleSizes = adapter.listHardwareProfiles();
+      assertFalse(Iterables.isEmpty(roleSizes));
+
+      for (RoleSize roleSize : roleSizes) {
+         assertNotNull(roleSize);
+      }
+   }
+
+   @AfterGroups(groups = "live")
+   protected void tearDown() {
+      if (deployment != null) {
+         adapter.destroyNode(deployment.getNodeId());
+      }
+      super.tearDown();
+   }
+
+   @Override
+   protected Iterable<Module> setupModules() {
+      return ImmutableSet.<Module> of(getLoggingModule(), new SshjSshClientModule());
+   }
+
+   @Override
+   protected Properties setupProperties() {
+      Properties properties = super.setupProperties();
+      properties.setProperty("jclouds.ssh.max-retries", "10");
```

Comment:
Users should be able to override this by setting the value from the command line. Set it only if not present.